hub/core/compute/provider.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+
+
+# TODO, will probably also need Queue to get tqdm working properly
+class ComputeProvider(ABC):
+    """An abstract base class for implementing a compute provider."""
+
+    def __init__(self, workers):
+        self.workers = workers
+
+    @abstractmethod
+    def map(self, func, iterable):
```

Comment:
This is confusing with the python `map`, which returns an iterable and does no work. I think `apply` is a more suitable name.

Reply:
The reason for using map is that it directly corresponds to 'map' of most famous libararies so adding a new one is obvious. pathos' Process pool and thread pool have map, so does Ray.